tools/format_python_tools.py

```diff
@@ -0,0 +1,60 @@
+import argparse
+import fnmatch
+import os
+import sys
+
+from yapf.yapflib.yapf_api import FormatFile
+
+
+def collect_files():
+  """Collect all Python files in the tools directory."""
+  # TODO: Add ability to collect a specific file or files.
+  matches = []
+  for root, dirnames, filenames in os.walk('.'):
+    for filename in fnmatch.filter(filenames, '*.py'):
+      matches.append(os.path.join(root, filename))
+  return matches
+
+
+def validate_format(fix=False):
+  """Check the format of python files in the tools directory.
+
+    Arguments:
+        fix: a flag to indicate if fixes should be applied.
+    """
+  fixes_required = False
+  failed_update_files = set()
+  successful_update_files = set()
+  for file in collect_files():
+    reformatted_source, encoding, changed = FormatFile(
+        file, style_config='.style.yapf', in_place=fix, print_diff=not fix)
+    if not fix:
+      fixes_required = changed
```

Comment:
Should this be:
```
if changed:
  fixes_required = True
```
to avoid resetting this later when a file doesn't need fixes?